### ISSUE TITLE
Background color selection in warp_perspective() function

### ISSUE THREAD

**Issue by arubior** [kornia/kornia#901]
## 🚀 Feature

Possibility to select the color (RGB) for the background pixels in the `warp_perspective()` function (pixels whose color des not come from the original warped image).

## Motivation

OpenCV's [`warpPerspective()`](https://docs.opencv.org/master/da/d54/group__imgproc__transform.html#gaf73673a7e8e18ec6963e3774e6a94b87) function has a `borderValue` parameter allowing to select the color of the background pixels when the `borderMode` is set to `BORDER_CONSTANT`. This is very useful when the desired background color is not black (default). Currently Kornia's [`warp_perspective()`](https://kornia.readthedocs.io/en/latest/geometry.transform.html) only supports black (`zeros` option for `border_mode`).

## Pitch

I'd like to have a parameter similar to OpenCV's `borderValue` to select the color of the background when the `border_mode` of the `warp_perspective` function in Kornia is set to `zeros` (and in that case, probably this name should change from `zeros` to something like `constant`, being black the default color).

**Comment by edgarriba** [comment]
hi @arubior thanksl for reporting this. In fact, we should do support what you suggest through `border_mode` since it's internally mapped to `padding_mode` for `F.grid_sample`. Could you verify that ?

See also that's there's a small change in this coming PR: 
https://github.com/kornia/kornia/pull/896/files#diff-393441a973b582a54d0dc25e96233c9d5bf2e8d07dc21597cd2de3c96f386484R106

**Comment by arubior** [comment]
Hi @edgarriba thanks for your answer. 

Thanks for pointing that out, now I see that it's internally mapped to `padding_mode` of `F.grid_sample`, but as I see [in their documentation](https://pytorch.org/docs/stable/nn.functional.html), they only support `'zeros' | 'border' | 'reflection'`, am I right?

**Comment by edgarriba** [comment]
true - there's a missing feature from pytorch side here in `grid_sample`. I think it's worth opening a feature request there. In meantime, I believe we could hack this around with masks and blending.

**Comment by arubior** [comment]
Cool, I was thinking on handling it on my side with masks, but if you include it as a feature in kornia it'd be great.

**Comment by edgarriba** [comment]
It's not a priority right now - but I'll mark it as future future.

**Comment by dribnet** [comment]
Being able to specify a padding mode with the same semantics as "zeros" but instead allowing a custom RGB triple would be incredibly useful for me. If someone wants to point me to where this would live in the code base I'd be happy to take a crack at it.

**Comment by edgarriba** [comment]
@dribnet I believe this should happen right after this call
https://github.com/kornia/kornia/blob/master/kornia/geometry/transform/imgwarp.py#L117

Ideally this feature should be coming from pytorch `grid_sample` /cc @fmassa

In the meantime as @arubior suggested we could implement somehow with masks. What API do you find more suitable for this feature - If ```mode == "constant"``` we require to provide a float or tuple[float,float,float] or tensor (in case we want to specify a different background color per image in the batch) ? /cc @ducha-aiki
